telcell/pipeline.py

```diff
@@ -0,0 +1,57 @@
+from pathlib import Path
+from typing import Any, Iterable, Mapping, Sequence, Tuple, Union
+
+import lir.plotting
+import numpy as np
+
+from telcell.data.models import Track, is_colocated
+from telcell.models import Model
+
+
+def run_pipeline(
+        data: Iterable[Tuple[Track, Track, Mapping[str, Any]]],
+        models: Iterable[Model],
```

Comment:
Ik verwacht hier in `models` geen iterable maar een `Model`. Je zou natuurlijk meerdere models willen kunnen toetsen, maar er kunnen ook andere variabelen zijn (of komen) die je wil kunnen varieren, zoals bijvoorbeeld de periode van vergelijking. Dan wordt dit een heel complexe functie die vele taken krijgt: moet itereren over meerdere variabelen en vervolgens de pipeline runnen en de resultaten agregeren.

Reply:
Zie ook de grote discussie in de TODO binnen die functie. Wat je dan namelijk wel krijgt, is dat als je meerdere modellen op dezelfde data (dus met exact dezelfde data cruncher) wil runnen, dat je meerdere keren over dezelfde data moet itereren en daarmee of (1) de data in z'n geheel in geheugen moet houden wat erg inefficient kan zijn als je slicet o.b.v. dag en per slice de overige historie ook wil gebruiken, of (2) de data meerdere keren parsen wat mogelijk een dure operatie is.

Reply:
Ok. Als argument voor het later itereren over data zou ik dan willen toevoegen dat het beter past in het scheiden van verantwoordelijkheden.

Reply:
Kan je dat wat verder toelichten? Als in, moet er dan nu nog wat aan de code veranderd worden voor dit PR?

Reply:
Ik zou pleiten voor dezelfde indeling als authorship en lrbenchmark. Die hebben 1 functie die een specifieke configuratie (1 model) runt en resultaten (lrs, cllr etc) teruggeeft. Ook 1 plek waar de resultaten verzameld worden zodat je kan plotten over verschillende configuraties, en vanuit run.py wordt de boel aanstuurt: de set aan mogelijke parameter values bepalen en de classes initialiseren
- 1 functie die

Reply:
Lijkt me goed om dit op te pakken op het moment dat we `lrbenchmark` toe gaan voegen als dependency en dit PR voor nu te mergen, anders blijft het nog veel langer open staan.